mica/archive/aca_dark/dark_cal.py

```diff
@@ -30,21 +31,35 @@ def date_to_dark_id(date):
     """
     Convert ``date`` to the corresponding YYYYDOY format for a dark cal identifiers.
 
-    :param date: any DateTime compatible format
+    :param date: any CxoTime compatible format
     :returns: dark id (YYYYDOY)
     """
-    date = DateTime(date).date
-    return date[:4] + date[5:8]
+    time = CxoTime(date)
+    date_str = time.date
+    if not time.shape:
+        return date_str[:4] + date_str[5:8]
+    date_str = np.atleast_1d(date_str)
+    b = date_str.view((str, 1)).reshape(-1, date_str.dtype.itemsize // 4)
```

Comment:
I think these don't pass Tom's "no 1-letter variable names" feedback in this PR.